backend/src/disk/fsck.rs

```diff
@@ -39,21 +41,38 @@ pub async fn e2fsck_preen(
     e2fsck_runner(Command::new("e2fsck").arg("-p"), logicalname).await
 }
 
+fn backup_existing_undo_file<'a>(path: &'a Path) -> BoxFuture<'a, Result<(), Error>> {
```

Comment:
:question:  why didn't you make this fn async?

Reply:
it is async, but recursive, so it needs to be boxed

Reply:
Glad I asked, now I learned something